chain/chain/src/doomslug.rs

```diff
@@ -273,35 +236,46 @@ impl DoomslugApprovalsTrackersAtHeight {
         &mut self,
         now: Instant,
         approval: &Approval,
-        stakes: &Vec<ValidatorStake>,
+        stakes: &Vec<ApprovalStake>,
         threshold_mode: DoomslugThresholdMode,
     ) -> DoomslugBlockProductionReadiness {
-        if let Some(last_parent_hash) = self.last_approval_per_account.get(&approval.account_id) {
+        if let Some(last_parent) = self.last_approval_per_account.get(&approval.account_id) {
             let should_remove = self
                 .approval_trackers
-                .get_mut(&last_parent_hash)
+                .get_mut(last_parent)
                 .map(|x| {
                     x.withdraw_approval(&approval.account_id);
                     x.witness.is_empty()
                 })
                 .unwrap_or(false);
 
             if should_remove {
-                self.approval_trackers.remove(&last_parent_hash);
+                self.approval_trackers.remove(&last_parent);
             }
         }
-        self.last_approval_per_account.insert(approval.account_id.clone(), approval.parent_hash);
+
+        let account_id_to_stakes = stakes
+            .iter()
+            .map(|x| (x.account_id.clone(), (x.stake_this_epoch, x.stake_next_epoch)))
+            .collect::<HashMap<_, _>>();
+
+        assert_eq!(account_id_to_stakes.len(), stakes.len());
+
+        if !account_id_to_stakes.contains_key(&approval.account_id) {
+            return DoomslugBlockProductionReadiness::NotReady;
```

Comment:
why do we not store this approval?

Reply:
We don't need to store an approval from someone who's not supposed to have an approval